src/EPR.PRN.Backend.Obligation/Services/ObligationCalculatorService.cs

```diff
@@ -97,17 +105,76 @@ public async Task SaveCalculatedPomDataAsync(List<ObligationCalculation> calcula
             await _obligationCalculationRepository.AddObligationCalculation(calculations);
         }
 
-        public async Task<List<ObligationCalculationDto>?> GetObligationCalculationByOrganisationId(int id)
+        public async Task<List<PrnDataDto>?> GetObligationCalculation(Guid organisationId, int year)
         {
-            var result = await _obligationCalculationRepository.GetObligationCalculationByOrganisationId(id);
+            var prnDataCollection = new List<PrnDataDto>();
+            var materials = await _materialRepository.GetAllMaterials();
+            if (!materials.Any())
+            {
+                _logger.LogError("No Materials found in PRN BAckend Database");
+                return prnDataCollection;
+            }
+            var materialsWithRemelt = AddGlassRemelt(materials.ToList());
+            var obligationCalculations = await _obligationCalculationRepository.GetObligationCalculation(organisationId, year);
+            var acceptedTonnageForPrns = await _prnRepository.GetSumOfTonnageForMaterials(organisationId, EprnStatus.ACCEPTED.ToString());
+            var awaitingAcceptanceForPrns = await _prnRepository.GetSumOfTonnageForMaterials(organisationId, EprnStatus.AWAITINGACCEPTANCE.ToString());
+            var materialNames = materialsWithRemelt.Select(material => material.MaterialName);
+            foreach (var materialName in materialNames)
+            {
+                var obligationCalculation = obligationCalculations.Find(x => x.MaterialName == materialName);
+                var tonnageAccepted = GetTonnage(materialName, acceptedTonnageForPrns);
+                var tonnageAwaitingAcceptance = GetTonnage(materialName, awaitingAcceptanceForPrns);
+                var tonnageOutstanding = GetTonnageOutstanding(obligationCalculation?.MaterialObligationValue, tonnageAccepted);
+                prnDataCollection.Add(new PrnDataDto
+                {
+                    OrganisationId = organisationId,
+                    MaterialName = materialName,
+                    ObligationToMeet = obligationCalculation?.MaterialObligationValue,
+                    TonnageAccepted = tonnageAccepted ?? 0,
+                    TonnageAwaitingAcceptance = tonnageAwaitingAcceptance ?? 0,
+                    TonnageOutstanding = tonnageOutstanding,
+                    Status = GetStatus(obligationCalculation?.MaterialObligationValue, tonnageAccepted)
+                });
+            }
+            return prnDataCollection;
+        }
+
+        private static List<Material> AddGlassRemelt(List<Material> materials)
```

Comment:
This parameter can be either **ref** or **out** type so that we don't have to return and handle it by the caller

Reply:
We've agreed not proceed as it adds complications to the calling code in one instance.